ppa/archive/management/commands/index.py

```diff
@@ -137,18 +131,17 @@ def handle(self, *args, **kwargs):
 
         self.solr.commit(self.solr_collection)
 
-    def index(self, index_data):
+    def index(self, index_data, progbar=None):
         '''index an iterable into the configured solr instance
         and solr collection'''
 
         # NOTE: currently no good way to catch a connection
         # error when Solr is not running because we get multiple
         # connections during handling of exceptions.
         try:
-            self.solr.index(self.solr_collection, index_data,
-                            params=self.solr_index_opts)
+            # index in chunks and update progress bar if there is one
+            return Indexable.index_items(index_data, progbar=progbar)
         except Exception as err:
         # except (ConnectionError, RequestException) as err:
             # NOTE: this is still pretty ugly; what part should we return?
```

Comment:
does this still need to be resolved, or is the output OK?

Reply:
I think the output is good enough - at least it stops the script instead of repeatedly generating errors.  Maybe something we can keep an eye on as we refine this code.  I'll add some comments to clarify.